hw/xbox/nv2a/nv2a.h

```diff
@@ -440,9 +440,42 @@ typedef struct NV2ABlockInfo {
 extern const struct NV2ABlockInfo blocktable[];
 extern const int blocktable_len;
 
+DMAObject nv_dma_load(NV2AState *d, hwaddr dma_obj_address);
+void *nv_dma_map(NV2AState *d, hwaddr dma_obj_address, hwaddr *len);
+void nv2a_init(PCIBus *bus, int devfn, MemoryRegion *ram);
 void pgraph_init(NV2AState *d);
 void *pfifo_puller_thread(void *opaque);
 void pgraph_destroy(PGRAPHState *pg);
 void update_irq(NV2AState *d);
 
+#define DEFINE_PROTO(prefix)                                              \
+    uint64_t prefix##_read(void *opaque, hwaddr addr, unsigned int size); \
+    void prefix##_write(void *opaque,                                     \
+                        hwaddr addr,                                      \
+                        uint64_t val,                                     \
+                        unsigned int size);
+
+DEFINE_PROTO(pmc)
+DEFINE_PROTO(pbus)
+DEFINE_PROTO(pfifo)
+DEFINE_PROTO(prma)
+DEFINE_PROTO(pvideo)
+DEFINE_PROTO(ptimer)
+DEFINE_PROTO(pcounter)
+DEFINE_PROTO(pvpe)
+DEFINE_PROTO(ptv)
+DEFINE_PROTO(prmfb)
+DEFINE_PROTO(prmvio)
+DEFINE_PROTO(pfb)
+DEFINE_PROTO(pstraps)
+DEFINE_PROTO(pgraph)
+DEFINE_PROTO(pcrtc)
+DEFINE_PROTO(prmcio)
+DEFINE_PROTO(pramdac)
+DEFINE_PROTO(prmdio)
+// DEFINE_PROTO(pramin)
+DEFINE_PROTO(user)
+
+#undef DEFINE_PROTO
+
```

Comment:
Is there any reason why we'd need these prototypes in the header?
I think hiding them in the source code was a better choice.